File: Python/Warmup-1/front3.py
import unittest
'''

Given a string, we'll say that the front is the first 3 chars of the string. If the string length is less than 3, the front is whatever is there. 
Return a new string which is 3 copies of the front.


front3('Java')         -> 'JavJavJav'
front3('Chocolate')    -> 'ChoChoCho'
front3('abc')          -> 'abcabcabc'

'''

def front3(str):
 if len(str)<1:
  return str
  
 else:
  char3 = str[0:3]
  return 3*char3
      


class Test_front3(unittest.TestCase):

    def test_1(self):
        self.assertEqual(front3('Java'), 'JavJavJav')

    def test_2(self):
        self.assertEqual(front3('Chocolate'), 'ChoChoCho')

    def test_3(self):
        self.assertEqual(front3('abc'), 'abcabcabc')


if __name__ == '__main__':
    unittest.main() 

File: Java/Warmup_1/front3.java
/**
 * @author 	Luis Teixeira
 *
 * @data 	30 May 2018
 *
 * @mail	filipe.teixeira.996@gmail.com 
 */
package Warmup_1;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

/*

Given a string, we'll say that the front is the first 3 chars of the string.
If the string length is less than 3, the front is whatever is there.
Return a new string which is 3 copies of the front.


front3("Java") --> "JavJavJav"
front3("Chocolate") --> "ChoChoCho"
front3("abc") --> "abcabcabc"

 */
public class front3 {

	private static int i = 1;
	
	@Test
	public void Test1()
	{	
		System.out.println("**********Test "+i+"**********");
		System.out.println("front3(\"Java\") --> \"JavJavJav\"\n");
				
		assertEquals("front3(\"Java\") --> \"JavJavJav\"", "JavJavJav", front3("Java"));
		i++;
	}
	
	@Test
	public void Test2()
	{	
		System.out.println("**********Test "+i+"**********");
		System.out.println("front3(\"Chocolate\") --> \"ChoChoCho\"\n");
				
		assertEquals("front3(\"Chocolate\") --> \"ChoChoCho\"", "ChoChoCho", front3("Chocolate"));
		i++;
	}
	
	@Test
	public void Test3()
	{	
		System.out.println("**********Test "+i+"**********");
		System.out.println("front3(\"abc\") --> \"abcabcabc\"\n");
				
		assertEquals("front3(\"abc\") --> \"abcabcabc\"", "abcabcabc", front3("abc"));
		i++;
	}
	
	
	public String front3(String str) {
		 
		if (str.length() < 3) {
			return str+str+str;
		}
		
		String ret = str.substring(0, 3);
		
		return ret+ret+ret;
		
	}

	
}
